scalapb/scalapb_2.13/src/main/scala/com/linecorp/armeria/common/scalapb/ScalaPbJsonMarshaller.scala

```diff
@@ -39,13 +39,16 @@ final class ScalaPbJsonMarshaller private (
     jsonParser: Parser = jsonDefaultParser
 ) extends GrpcJsonMarshaller {
 
-  override def serializeMessage[A](marshaller: Marshaller[A], message: A, os: OutputStream): Unit = {
-    if (!message.isInstanceOf[GeneratedMessage])
-      throw new IllegalStateException(
-        s"Unexpected message type: ${message.getClass} (expected: ${classOf[GeneratedMessage]})")
-    val msg = message.asInstanceOf[GeneratedMessage]
-    os.write(jsonPrinter.print(msg).getBytes())
-  }
+  override def serializeMessage[A](marshaller: Marshaller[A], message: A, os: OutputStream): Unit =
+    message match {
+      case msg: GeneratedSealedOneof =>
+        os.write(jsonPrinter.print(msg.asMessage).getBytes())
```

Comment:
Awesome! Could you add tests for verifying that Armeria server and client correctly marshaling oneof?